From 7a426c581f02c428f0dcc65b323e4ecfa7d40d56 Mon Sep 17 00:00:00 2001
From: Ilya Leoshkevich <iii@linux.ibm.com>
Date: Thu, 26 Jan 2023 00:38:15 +0100
Subject: [PATCH] [mono] Build mono logging profiler on s390x (#72049)

At the moment we don't have too many profiling options on s390x, and
the logging profiler comes in handy. Enable building it on s390x and
define MONO_DLL_EXPORT in order to unhide mono_profiler_init_log().
---
 src/mono/mono/profiler/CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/mono/mono/profiler/CMakeLists.txt b/src/mono/mono/profiler/CMakeLists.txt
index 784ceb0203d..de72e324b15 100644
--- a/src/mono/mono/profiler/CMakeLists.txt
+++ b/src/mono/mono/profiler/CMakeLists.txt
@@ -10,9 +10,10 @@ include_directories(
   ${PROJECT_SOURCE_DIR}/../sgen)
 
 if(NOT DISABLE_LIBS)
-  if(HOST_ANDROID OR HOST_IOS OR HOST_TVOS)
-    # Build the logging profiler only for mobile platforms
+  if(HOST_ANDROID OR HOST_IOS OR HOST_TVOS OR HOST_S390X)
+    # Build the logging profiler only for certain platforms
     add_library(mono-profiler-log SHARED helper.c log.c log-args.c)
+    target_compile_definitions(mono-profiler-log PRIVATE -DMONO_DLL_EXPORT)
     target_link_libraries(mono-profiler-log monosgen-shared eglib_objects)
     if(HOST_ANDROID)
       target_link_libraries(mono-profiler-log log)
-- 
2.43.0

